providers/implementations/kem/rsa_kem.c

```diff
@@ -286,6 +286,12 @@ static int rsasve_generate(PROV_RSA_CTX *prsactx,
             *secretlen = nlen;
         return 1;
     }
+
+    if (outlen != NULL && *outlen < nlen) {
+        ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_OUTPUT_LENGTH);
+        return 0;
+    }
+
```

Comment:
This added condition is not documented. I agree with adding it but please also adjust the documentation accordingly.

Reply:
docs added